edb/server/cluster.py

```diff
@@ -163,28 +137,53 @@ def init(self, *, server_settings=None):
 
         self._init(self._pg_cluster)
 
-    def start(self, wait=60, **settings):
-        port = settings.pop('port', None) or self._port
-        if port == 'dynamic':
-            port = find_available_port()
+    def start(self, wait=60, *, port: int=None, **settings):
+        if port is None:
+            port = self._port
 
-        self._effective_port = port
+        if port == 0:
+            cmd_port = 'auto'
+        else:
+            cmd_port = str(port)
 
         extra_args = ['--{}={}'.format(k.replace('_', '-'), v)
                       for k, v in settings.items()]
-        extra_args.append('--port={}'.format(self._effective_port))
+        extra_args.append(f'--port={cmd_port}')
+        if port == 0:
+            extra_args.append('--echo-runtime-info')
 
+        env: typing.Optional[dict]
         if self._env:
             env = os.environ.copy()
             env.update(self._env)
         else:
             env = None
 
+        stdout = subprocess.DEVNULL
+        if port == 0:
+            stdout = subprocess.PIPE
+
         self._daemon_process = subprocess.Popen(
             self._edgedb_cmd + extra_args,
-            stdout=sys.stdout, stderr=sys.stderr,
-            env=env)
+            stdout=stdout,
+            stderr=subprocess.DEVNULL,
```

Comment:
Why is stderr going to /dev/null?

Reply:
I'm not sure it's right to pipe it into sys.stderr. It was like that before and didn't help me debug while working on this PR one bit. I can revert to sys.stderr if you want.